src/clr-angular/data/datagrid/providers/detail.service.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+@Injectable()
+export class DetailService {
+  private toggleState = null;
+  protected _state: BehaviorSubject<boolean | null> = new BehaviorSubject<boolean | null>(this.toggleState);
+  private cache: any;
+
+  public get stateChange(): Observable<boolean | null> {
+    return this._state.asObservable();
+  }
+
+  close() {
+    this.toggleState = false;
+    this._state.next(this.toggleState);
+  }
+
+  open(item: any) {
+    this.cache = item;
+    this.toggleState = true;
+    this._state.next(this.toggleState);
+  }
+
+  toggle(item: any) {
+    if (this.toggleState && this.cache !== item) {
+      this.open(item);
```

Comment:
Here we directly jump to another element, thus we omit the firing of the `stateChange(false)`, which leads to omitting         `this.columnsService.resetToLastCache();` in `MainRenderer#toggleDetailPane()`, which I guess leads to the following problem:
- open detal
- open other detail
- close detail
- observe columns - only the 1st is present though we don't have details open.
I may be wrong this is the reason, but the problem is reproducible.

Reply:
Thanks, will review this.